docs/ticket-triage.rst

```diff
@@ -0,0 +1,27 @@
+=============
+Ticket Triage
+=============
+
+Ticket triage is a weekly, informal ritual that is meant to solve the problem of
+tickets getting opened and then forgotten about. It is simple and keeps project
+momentum going and prevents ticket cruft.
+
+It fosters conversation around project tasks and philosophies as they relate to 
+milestones.
+
+Process
+-------
+- The role of Ticket Triager rotates weekly 
+- The Triager needs admin status on ``Trac``
+- The Triager looks at all the tickets that have been created in the last week 
+    - They can use a custom query or do this as the week progresses 
+    - BONUS ROUND: Dig up a stale ticket from the past
+- Assign each ticket to a milestone on the Roadmap 
+- The following situations merit discussion:
+    - A ticket doesn't have an appropriate milestone and we should create one 
+    - A ticket should probably be deleted
```

Comment:
I don't think Trac has a provision to delete tickets.  A ticket's status can be changed to invalid, duplicate, won't fix, somebody else's problem, etc.

Reply:
Cool, thanks for spotting that! I didn't even think to look for warnings :) A bad habit - caring only about errors and forgetting about warnings ;)

About deletion: I made and deleted a dummy ticket just now. But, the greater discussion is whether we want to delete tickets. I think it's perfectly okay, but if people want to do something that is more history-preserving like change its status, that's okay w/me too. 

I favor deletion because 

1) Some tickets are no longer relevant because perhaps there's a speculative feature that would not be feasible or useful to add, or they they don't have anything actionable or other vital content in them.
2) if something is really important, then a person will make a ticket again. I would guess that there are already duplicated or duplicate-like tickets in Trac.
3) My sense is that we'd be relatively conservative in choosing to delete tickets.

![Screenshot from 2021-04-05 00-32-50](https://user-images.githubusercontent.com/12126401/113523230-d2485e80-95a6-11eb-9d76-059ece8ed9a5.png)

Reply:
> Cool, thanks for spotting that! I didn't even think to look for warnings :) A bad habit - caring only about errors and forgetting about warnings ;) I have fixed it.

It is really my fault.  I should have thought about this when adding that CI step. :-)

> About deletion: I made and deleted a dummy ticket just now. But, I think perhaps the greater discussion is whether we want to delete tickets. I think it's perfectly okay, but if people want to do something that is more history-preserving like change its status, that's okay w/me too.
> 
> I favor deletion because
> 
>     1. Some tickets are no longer relevant because perhaps there's a speculative feature that would not be feasible or useful to add, or they they don't have anything actionable or other vital content in them.
> 
>     2. if something is really important, then a person will make a ticket again. I would guess that there are already duplicated or duplicate-like tickets in Trac.
> 
>     3. My sense is that we'd be relatively conservative in choosing to delete tickets.

Ah I didn't know it was possible to delete Trac tickets.  Thanks!

There's also a question of etiquette when removing other people's words though. If someone has spent some of their time and effort on writing a ticket, in general it is better to be respectful of that.

If I have made some silly mistake when filing a ticket, I can probably delete it afterwards without causing much harm, before anyone else has added their thoughts to it.   Once someone else has chimed in with a comment or some sort of useful feedback, I will have to be careful about deleting that ticket.

Likewise I wouldn't want to delete a ticket someone else has created, unless it contains some kind of abuse or spam.  We do not have that problem right now though. :-)

In case 1, a better resolution would be communicating why something can't be done.  There likely won't be a case 2, if we delete tickets.  They're likely to silently go away, or go badmouth us elsewhere.

Reply:
> 
> It is really my fault. I should have thought about this when adding that CI step. :-)
> 

No worries :) Like I said, I generally ignore warnings unless I know what to do with them. I am used to `rspec`/`rake` in `ruby`, which looks like this:

![image](https://user-images.githubusercontent.com/12126401/113568935-d5c4ff80-9611-11eb-8f31-c4be1cd56dc0.png)

> Likewise I wouldn't want to delete a ticket someone else has created, unless it contains some kind of abuse or spam. We do not have that problem right now though. :-)
> 

Okay, I'll reword the line to avoid deletion as a form of resolution.